microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks.md

```diff
@@ -40,131 +55,230 @@ As explained in [Fix memory problems](index.md), memory issues affect page perfo
+1. In the **Detached Elements** feature, click the **Collect garbage** (![The 'Collect garbage' icon](./dom-leaks-images/collect-garbage-icon.png)) icon.
 
-For more information about heap snapshots, read [Record heap snapshots using the Memory tool](heap-snapshots.md).
+1. Click the **Get Detached Elements** (![The Get Detached Elements icon](./dom-leaks-images/get-detached-elements-icon.png)) icon.
 
-To identify the JavaScript code that references a detached element:
+   <!-- todo: what's the new, desired, beneficial state?  what have we accomplished now? -->
+   You can now see XYZ.   Next, identify the JavaScript code that references a particular detached element, as follows.
 
-1. In the **Detached Elements** tool, click the **Analyze** (![The Analyze icon](./dom-leaks-images/analyze-icon.png)) icon.
+1. In the **Detached Elements** feature, click the **Analyze** (![The Analyze icon](./dom-leaks-images/analyze-icon.png)) icon.
 
    The **Memory** tool opens in the **Quick View** toolbar, at the bottom of DevTools:
 
    ![Analyze Detached Elements in the Detached Elements tool](./dom-leaks-images/analyze-detached-elements.png)
 
-1. From the **Detached Elements** tool, select the **Id** field of a detached element.
+1. In the **Detached Elements** tool, in the **Id** column, click **@21299**, which is the ID of one of the `<div class="message">` elements.
 
-   The **Memory** tool automatically selects the object in the heap that is referencing the detached element. We call these objects **Retainers**:
+   The **Memory** tool automatically selects the object in the heap that is referencing the detached element.  These objects are called _retainers_, and are listed in the **Retainers** tab:
 
    ![Referencing a heap snapshot from the Detached Elements tool](./dom-leaks-images/heap-snapshot.png)
 
-1. In the **Memory** tool, select the link **room.js:13**.
+1. In the **Memory** tool, in the **Retainers** tab, in the **[2] in Array @58525** retainer entry, in the **unmounted in Room @57707** retainer sub-entry, click the link **room.js:13**.
 
-   The **Sources** tool opens in the **Activity Bar** and shows line 13 of the file **room.js**.
+   The **Sources** tool opens in the **Activity Bar** and shows line 13 of the file **room.js**:
 
-1. In the `hide()` function of **room.js**, the JavaScript code of the demo application adds each message in the room to an `unmounted` array. The `unmounted` array is the object that is referencing the detached element:
+   ![Line 13 in room.js](./dom-leaks-images/room-js-line-13.png)<!-- todo: redo png to show line 13 -->
 
-   ![Identifying the JavaScript that is retaining the detached element](./dom-leaks-images/javascript-retainers.png)
+1. Scroll down to line 49:<!-- todo: why? -->
```

Comment:
why did we scroll from line 13 to line 49?  (in step 15)
I'm having to add a screenshot showing line 13 that you actually end up at.  (why isn't that line shown? can't understand, w/o showing it)
@captainbrosset

https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks?branch=pr-en-us-3282&tabs=detached-elements-tool

Reply:
added line 13 screenshot, some improvement of explaining lines of code